cast/src/lib.rs

```diff
@@ -1254,45 +1255,37 @@ impl SimpleCast {
                 let client = Client::new(chain, api_key)?;
 
                 // get the source
-                let contract_source = match client.contract_source_code(address).await {
-                    Ok(src) => src,
-                    Err(ethers_etherscan::errors::EtherscanError::InvalidApiKey) => {
-                        eyre::bail!("Invalid Etherscan API key. Did you set it correctly? You may be using an API key for another Etherscan API chain (e.g. Ethereum API key for Polygonscan).")
+                let source = match client.contract_source_code(address).await {
+                    Ok(source) => source,
+                    Err(EtherscanError::InvalidApiKey) => {
+                        eyre::bail!("Invalid Etherscan API key. Did you set it correctly? You may be using an API key for another Etherscan API chain (e.g. Etherscan API key for Polygonscan).")
+                    }
+                    Err(EtherscanError::ContractCodeNotVerified(address)) => {
+                        eyre::bail!("Contract source code at {:?} on {} not verified. Maybe you have selected the wrong chain?", address, chain)
                     }
                     Err(err) => {
                         eyre::bail!(err)
                     }
                 };
 
-                if contract_source
-                    .items
-                    .iter()
-                    .any(|item| item.abi == "Contract source code not verified")
-                {
-                    eyre::bail!("Contract source code at {:?} on {} not verified. Maybe you have selected the wrong chain?", address, chain)
-                }
-
-                let contract_source_names = contract_source
+                let names = source
                     .items
                     .iter()
                     .map(|item| item.contract_name.clone())
                     .collect::<Vec<String>>();
 
-                let mut abis = Vec::with_capacity(contract_source.items.len());
-                for item in &contract_source.items {
-                    abis.push(serde_json::from_str(&item.abi)?);
-                }
+                // TODO: Abi to RawAbi ?
```

Comment:
@mattsse not sure how to do this here, metadata's `abi` field is an `Abi` struct when we need a `RawAbi`, reasons explained in `foundry_utils::abi::abi_to_solidity`

Reply:
also suggesting that we do move RawAbi to ethers-core

I'd change the abi field to String and add helper functions that return RawAbi and Abi, I think